src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -254,7 +257,7 @@ private void requestBlockToBuildingIfRequired(BuildingBuilder building, IBlockSt
     /**
      * Adds entities to the builder building if he needs it.
      */
-    private void requestEntityToBuildingIfRequired(Template.EntityInfo entityInfo)
+    private void requestEntityToBuildingIfRequired(final Template.EntityInfo entityInfo)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)